resfs.c

```diff
@@ -0,0 +1,72 @@
+/* Copyright (C) 2023, Oracle and/or its affiliates. All rights reserved
+ *
+ * This file is part of libresource.
+ *
+ * libresource is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * libresource is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with libresource. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _RESOURCE_H
+#include "resource.h"
+#endif
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "resfs.h"
+#include "resource_impl.h"
+#include <errno.h>
+#include <libgen.h>
+
+static char buffer[4096];
+
+int getfsinfo(int res_id, void *out, size_t sz, void **hint, int flags)
+{
+	int ret;
+	unsigned long long *fs;
+
+	switch (res_id) {
+	case FS_AIONR:
+		ret = file_to_buf(AIONR, buffer, sizeof(buffer));
+		if (ret == -1)
+			return -1;
+		sscanf(buffer, "%Lu", (unsigned long long *) out);
```

Comment:
In all these cases, you should probably be checking the sscanf return value and forwarding an error on to the caller.